_posts/2021-09-13-changelog-155.markdown

```diff
@@ -0,0 +1,76 @@
+
+  * [#7772] Show warning/errors during fetching stdlib info in [Sync View](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-cargo-tool-window.html)
+
+* [#7777] Enable by default code completion ranking for Rust based on machine learning. Now the completion suggestions are sorted by their relevance based on the rules learned from data we have gathered anonymously during our EAPs. Note that we have not collected any source code, only information about your interactions with the code completion UI. In case of any issues, you can disable the new ranking model via `Editor | General | Code Completion | Machine Learning-Assisted Completion` settings. You can also see completion sorting changes by enabling `Mark position changes in the completion popup` in `Preferences | Editor | General | Code Completion`
+
+  <img src="/assets/posts/changelog-155/ml-completion.png" alt="ML completion sorting" width="550px">
+
+* [#7769], [#7751] Correctly parse `unsafe` keyword in extern and module blocks, annotate them as errors
+
+  <img src="/assets/posts/changelog-155/unsafe-extern-mod.png" alt="Parse and annotate unsafe in extern and module blocks" width="550px">
+
+* [#7734] Introduce new icons for macro items, providing `macro_rules!` and `macros 2.0` items with distinctive ones
+
+  <img src="/assets/posts/changelog-155/new-macro-icons.png" alt="New macro items icons" width="550px">
+
+* [#7663] Support `Move Element Left/Right` action for predicates in `where` clause
+
+  {% include gif-img.html path="/assets/posts/changelog-155/where-move" w="550px" %}
+
+* [#7757] Parse and annotate block expressions as default const generic values when using `const_generics_defaults` compiler feature
+
+* [#3790] Deprecated item inspection now correctly handles items that will be deprecated in a future version (by [@mibac138])
+
+## Fixes
+
+* [#7790] Generate [const generics](https://rust-lang.github.io/rfcs/2000-const-generics.html) by `Implement Members` quick-fix properly
+
+* [#7758] Don't fetch stdlib metadata for `rustc` project since it's already fetched during project structure setup
+
+* [#7752] Properly render [negative `impl`](https://doc.rust-lang.org/beta/unstable-book/language-features/negative-impls.html) items in `Go to implementation` popup
+
+* [#7728] Fix DTrace [profiler integration](https://plugins.jetbrains.com/plugin/8182-rust/docs/rust-profiler.html) with CLion 2021.2
+
+* [#7726] Correctly print restricted visibility of items in quick documentation instead of always printing "pub" (by [@afetisov])
```

Comment:
```suggestion
* [#7726] Correctly print restricted visibility of items in [quick documentation](https://plugins.jetbrains.com/plugin/8182-rust/custom-pages/docs/rust-code-reference-info.html#quick-docs) instead of always printing `pub` (by [@afetisov])
```